subsys/nrf_security/CMakeLists.txt

```diff
@@ -102,8 +102,29 @@ endif()
 
 set(CONFIG_MBEDTLS_PSA_CRYPTO_EXTERNAL_RNG  True)
 
+if(CONFIG_PSA_SSF_CRYPTO_CLIENT AND NOT CONFIG_NRF_SECURITY)
```

Comment:
So `CONFIG_PSA_SSF_CRYPTO_CLIENT` is independent from `CONFIG_NRF_SECURITY` but still using some of its CMake logic? I'm wondering if this can potentially cause some issues...

Reply:
The thing is that right now SSF is enabled with NRF_SECURITY. So that setup works currently. I didn't want to change the existing setup in this PR. What I did is to allow enabling it even when NRF_SECURITY is not used. This is useful for applications which just needs RNG for example and I didn't want to force enabling NRF_SECURITY for these yet. 

This should not be the final solution to this, this is basically the starting point to allow widely usage of RNG from Cracen. The next step on this would be that we refactor nrf_security to allow building with the SSF client with minimal configuration/code overhead.

Reply:
So if we allow enabling it even when `NRF_SECURITY` is not enabled, wouldn't it make sense to move the logic away from it? :thinking:

Reply:
I was considering this, but since I believe that soon we will refactor nrf_security to have minimal impact for this build configuration I decided to not separate them. That's because if nrf_security becomes more minimal we will  make this configuration a part of nrf_security again.